de.fhdo.lemma.technology.technologydsl.extractor/src/de/fhdo/lemma/technology/technologydsl/extractor/TechnologyDslExtractor.xtend

```diff
@@ -0,0 +1,95 @@
+package de.fhdo.lemma.technology.technologydsl.extractor
+
+import de.fhdo.lemma.technology.DataFormat
+import de.fhdo.lemma.technology.Protocol
+import de.fhdo.lemma.technology.Technology
+import de.fhdo.lemma.technology.CommunicationType
+import de.fhdo.lemma.technology.JoinPointType
+import de.fhdo.lemma.technology.TechnologySpecificPrimitiveType
+import de.fhdo.lemma.technology.ServiceAspect
+
+/**
+ * Model-to-text extractor for the Technology DSL.
+ *
+ * @author <a href="mailto:jonas.sorgalla@fh-dortmund.de">Jonas Sorgalla</a>
+ */
+class TechnologyDslExtractor {
+
+    /**
+     * Extract TechnologyModel
+     */
+    def String extractToString(Technology techModel) {
+        val types = if (techModel.primitiveTypes !== null) {
+            // Currently only supports primitiveTypes for generation
+            '''types {
+                «FOR pt : techModel.primitiveTypes SEPARATOR '\n'»«pt.extractToString»«ENDFOR»
+            }
+            '''
+        }
+        val protocols = if (techModel.protocols !== null) {
+            '''protocols {
+                «FOR pr : techModel.protocols SEPARATOR '\n'»«pr.extractToString»«ENDFOR»
+            }
+            '''
+        }
+        val aspects = if (techModel.serviceAspects !== null) {
+            '''service aspects {
+                «FOR sa : techModel.serviceAspects SEPARATOR '\n'»«sa.extractToString»«ENDFOR»
+            }
+            '''
+        }
+        '''technology «techModel.name» {
+            «types»
+            «protocols»
+            «aspects»
+        }
+        '''
+    }
+
+    /**
+     * Extract Protocol
+     */
+    def String extractToString(Protocol protocol) {
+        val comm = '''«IF (protocol.communicationType == CommunicationType.SYNCHRONOUS)»sync «ELSE
+            »async «ENDIF»'''
+
+        val formats = '''«FOR df : protocol.dataFormats SEPARATOR ', '»"«
+            df.extractToString»"«ENDFOR»'''
+
+        val defaultFormat = '''«IF (protocol.defaultFormat !== null)
+            » default with format "«protocol.defaultFormat.formatName»"«ENDIF»'''
+
+        '''«comm»«protocol.name» data formats «formats»«defaultFormat»;'''
+    }
+
+    /**
+     * Extract Aspect
+     */
+    def String extractToString(ServiceAspect aspect) {
+        '''aspect «aspect.name»«IF aspect.isIsSingleValued»<singleval>«ENDIF» for «
+            FOR jp : aspect.joinPoints SEPARATOR ', '»«jp.extractToString»«ENDFOR»;'''
+    }
+
+    /**
+     * Extract DataFormat
+     */
+    def String extractToString(DataFormat dataformat) {
+        '''«dataformat.formatName»'''
+    }
+
+    /**
+     * Extract JointPointType
+     */
+    def String extractToString(JoinPointType jpt) {
+        '''«jpt.literal.toLowerCase»'''
```

Comment:
Since there is no static template part, just make this `return jpt.literal.toLowerCase` or even better `return jpt.name.toLowerCase`, which seems more common in LEMMA's codebase.